### PR TITLE
Kyle/customers

### PR DESCRIPTION
**Pull request by kyleqhua**
🖋 PR Description
Customer page. Can search through records. Sort/Filter feature not implemented because need to look at Airtable date. Airtable integration in progress. Need a little css tweaks.

🤳Screenshots (if applicable)
ezgif-4-8176b295554a

🐞 Bug Report

🧪 Testing

### CODE REVIEW

**Comment by julianrkung** [comment]
Is this blocked on anything?

File: src/components/CustomerCard.tsx
Comment (julianrkung) [inline src/components/CustomerCard.tsx]: since `cardContent` is decently short, lets not abbreviate it so it's clearer. Took me a little bit to figure out what it meant.

File: src/components/HamburgerIcon.tsx
Comment (julianrkung) [inline src/components/HamburgerIcon.tsx]: Is this component used? I think it was deleted some time ago, so if you're not using it can you delete this file?

File: src/components/NavBar.tsx
Comment (julianrkung) [inline src/components/NavBar.tsx]: Is this component used? I think it was deleted some time ago and replaced with `BottomNavigation`. Can you check that it's no longer used in the most current version and delete it if it isn't?

File: src/components/Navigation.tsx
Comment (julianrkung) [inline src/components/Navigation.tsx]: Is this component used? I think it was deleted some time ago and replaced with BottomNavigation. Can you check that it's no longer used in the most current version and delete it if it isn't?

File: src/components/NavigationBar.js
Comment (julianrkung) [inline src/components/NavigationBar.js]: Is this component used? I think it was deleted some time ago and replaced with BottomNavigation. Can you check that it's no longer used in the most current version and delete it if it isn't?

File: src/styles/HamburgerStyle.ts
Comment (julianrkung) [inline src/styles/HamburgerStyle.ts]: I don't think this file is used anymore since Hamburger isn't. Can you double check and delete if its not?

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx]: Delete commented out code

File: src/components/hooks.ts
Comment (julianrkung) [inline src/components/hooks.ts]: Is this used anywhere? I grep'd for it and i can't find any uses. Delete file/function if not used.

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx]: Doesn't this grab the customers from all the sites the user has access to? It's my understanding that we only want the customers of the current site the user has set.

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx:97]: Can we get this `Sort By` working in this PR? Is there a reason it's placeholdered / on backburner?

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx]: We need to account for empty array case if customer has no meter readings

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx]: `customer` instead of `curr`. I also prefer `getLatestMeterReadingDate` to the current name since nothing is being calculated but that's small

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx:47]: Is this `fullCustomers` used anywhere? We might be able to get away with just `customers` (maybe `filteredCustomers` or some indication that this is the customers being shown is a better name), and the `allCustomersTrie` which will always hold all the customers afaik

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx:54]: Performance might suffer here since we call `getCustomers` (which iterates through all customers and does some sets) every time we do a rerender, which happens when we search. We might just want to call this `getCustomers` on the initial render to save all the extra work when we render during filters. Open to discuss on this though if you have opinions

**Review by julianrkung — CHANGES_REQUESTED** [review]
Can you delete the empty/outdated files in this PR?

Left some comments on code logic first. When you do that push for green icons I'll take another look and also comment on frontend (which should just be small changes).

File: src/components/HamburgerIcon.tsx
Comment (kyleqhua) [reply src/components/HamburgerIcon.tsx]: yeah i think its left over from old master

File: src/screens/Customers/CustomerMain.tsx
Comment (kyleqhua) [reply src/screens/Customers/CustomerMain.tsx]: wait yeah that prob more apporpriate in regards to the app.

File: src/screens/Customers/CustomerMain.tsx
Comment (kyleqhua) [reply src/screens/Customers/CustomerMain.tsx:97]: I think Jen originally gave the task to Emma? So I didn't implement it. I can do it if needed.

File: src/screens/Customers/CustomerMain.tsx
Comment (kyleqhua) [reply src/screens/Customers/CustomerMain.tsx:47]: I forgot that the trie holds all the customers. Thanks!

File: src/components/CustomerCard.tsx
Comment (julianrkung) [inline src/components/CustomerCard.tsx]: Replace with `body1` in `ThemeStyles.ts` so if it gets changed we don't have to change all individually

File: src/components/CustomerCard.tsx
Comment (julianrkung) [inline src/components/CustomerCard.tsx]: Replace with `body1` in `ThemeStyles.ts` so if it gets changed we don't have to change all individually

File: src/components/UserSearchBar.tsx
Comment (julianrkung) [inline src/components/UserSearchBar.tsx]: something related to `onSearchChange` might be more appropriate since `searchFun` isn't as descriptive as it should be imo. Try not to abbreviate into words like `fun` because it's harder to understand immediately what this is. Clarity in variable names is gonna be really important if NPO is going to be doing development after handoff

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx]: Replace with `h1` in `ThemeStyles.ts` so if it gets changed we don't have to change all individually

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx]: use `const` if the variable isn't gonna change

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx]: use `const` if the variable isn't gonna change

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [reply src/screens/Customers/CustomerMain.tsx:97]: Talked with Kyle offline. This'll be done in a subsequent PR.

**Review by julianrkung — CHANGES_REQUESTED** [review]
Looking good, probably just like ~1-2 more iterations!

Could you delete all the empty, outdated files too (look at PR diff to find them all)?

Also I found a bug. If you search on the page and then delete the search (so the search bar is empty, you end up with no customers. Looks like a change needs to be made to `handleSearchChange`
image


Last thing: The screen looks a little off on different phones (especially something like the `iPhone 6/7/8 Plus`. I think this has to do with the hardcoded pixel styling that's being used. I don't think we can trust/copy-paste the Figma-generated CSS for stuff like widths and margins etc. @jenmhoang did a good job building base components to use so that our screens look uniform. Look at how `CustomerProfile` and `Home` are structured (using `BaseScreen` etc...) to make sure that the page looks good on all platforms.

(this is a screenshot of how it looks on iphone plus sizes)
image

File: src/components/UserSearchBar.tsx
Comment (kyleqhua) [reply src/components/UserSearchBar.tsx]: yessir that makes sense.

File: src/screens/Customers/CustomerMain.tsx
Comment (kyleqhua) [reply src/screens/Customers/CustomerMain.tsx:47]: I remember why we needed full customers. There is no way to get the full customers from the trie. With an empty string '' the trie returns nothing. Looked fpr documentation and it seems like you cant get it all back. I looked up how the trie was used in Unloop and we used the same method (keep a full copy of list): 
https://github.com/calblueprint/unloop/blob/10b27cbc70fbc85d2322db00d3fc7088facd2713/app/javascript/components/StaffDashboard/index.js#L45

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx]: This causes awkward spacing between the search bar and title. See screenshots.

With:
image

Without:
image

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx:26]: i think this is default so there's no need to put `flexDirection: row`

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx]: Add `rightIcon user` to match Figma.

Right now:
image

Figma:
image

File: src/components/CustomerCard.tsx
Comment (julianrkung) [inline src/components/CustomerCard.tsx]: default flexDirection is row. can be removed

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx]: Component isn't being used so we can delete

File: src/components/CustomerCard.tsx
Comment (julianrkung) [inline src/components/CustomerCard.tsx]: Can you delete the unused imports here?

File: src/components/UserSearchBar.tsx
Comment (julianrkung) [inline src/components/UserSearchBar.tsx]: Can you delete the unused imports here?

File: src/components/UserSearchBar.tsx
Comment (julianrkung) [inline src/components/UserSearchBar.tsx:30]: According to Figma, border should be the `main` palette in `ThemeStyles.ts`  when active and otherwise `BDBDBD`. Lets put off the searchbar animation for another PR if we decide to do it since it's not that important.

Figma screenshot:
image

File: src/components/UserSearchBar.tsx
Comment (julianrkung) [inline src/components/UserSearchBar.tsx]: If the styles don't use theme, you don't need to have it as a parameter. In general, don't add a parameter unless it's going to be used (this might change since we should be using the `primary.main` from themeStyles.ts).

File: src/components/UserSearchBar.tsx
Comment (julianrkung) [inline src/components/UserSearchBar.tsx:13]: The search icon looked like it was a little offcenter from the search. To regulate you can restrict the dimensions to be proportional to searchWrapper. I put a `height: 75%` and it looks nicer. You can play around and see if there's anything better but feel free to use 75% otherwise.

Current:
image

Changed icon dimension:
image

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [reply src/screens/Customers/CustomerMain.tsx:47]: Sounds good, thanks for checking it! :^)

**Review by julianrkung — CHANGES_REQUESTED** [review]
Final comments. Should be good to merge after all these are addressed.

One last thing, it looks like scroll doesn't behave how it's supposed to on the page. I believe the top title / search bar / sort should be sticky on scroll. That's how it is on the `CustomerProfile.tsx` at least. Attached a couple clips to show what I mean.

Current / nonsticky header:
https://user-images.githubusercontent.com/35501399/104109947-b2471e80-5287-11eb-9254-9c6d275a4df4.mov

Sticky header / customer profile:
https://user-images.githubusercontent.com/35501399/104109950-bd9a4a00-5287-11eb-83c5-d6f112988225.mov

**Review by julianrkung — APPROVED** [review]
Thanks for working so hard on this! :^)

**Pull request merged**
